apollo-federation/src/merger/merger.rs

```diff
@@ -992,12 +992,520 @@ impl Merger {
+                static PERCENT_REGEX: LazyLock<regex::Regex> = LazyLock::new(|| {
+                    regex::Regex::new(r"^percent\((\d{1,2}(\.\d{1,8})?|100)\)$").unwrap()
+                });
+
+                let is_valid = if LABEL_REGEX.is_match(&override_label) {
+                    true
+                } else if let Some(captures) = PERCENT_REGEX.captures(&override_label) {
+                    if let Some(percent_str) = captures.get(1) {
+                        if let Ok(percent) = percent_str.as_str().parse::<f64>() {
+                            (0.0..=100.0).contains(&percent)
+                        } else {
+                            false
+                        }
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                };
+
+                if is_valid {
+                    result.set_override_label(idx, override_label.clone());
+                    result.set_override_label(from_idx, override_label.clone());
+                } else {
+                    self.error_reporter.add_error(CompositionError::OverrideLabelInvalid {
+                        message: format!(
+                            "Invalid @override label \"{}\" on field \"{}\" on subgraph \"{}\": labels must start with a letter and after that may contain alphanumerics, underscores, minuses, colons, periods, or slashes. Alternatively, labels may be of the form \"percent(x)\" where x is a float between 0-100 inclusive.",
+                            override_label, dest, subgraph_name
+                        ),
+                    });
+                }
+
+                // Add migration hint if label is valid
+                if is_valid {
+                    let message = if overridden_field_is_referenced {
+                        format!(
+                            "Field \"{}\" on subgraph \"{}\" is currently being migrated via progressive @override. It is still used in some federation directive(s) (@key, @requires, and/or @provides) and/or to satisfy interface constraint(s). Once the migration is complete, consider marking it @external explicitly or removing it along with its references.",
+                            dest, source_subgraph_name
+                        )
+                    } else {
+                        format!(
+                            "Field \"{}\" is currently being migrated with progressive @override. Once the migration is complete, remove the field from subgraph \"{}\".",
+                            dest, source_subgraph_name
+                        )
+                    };
+
+                    self.error_reporter.add_hint(CompositionHint {
+                        code: HintCode::OverrideMigrationInProgress.code().to_string(),
+                        message,
+                        locations: Default::default(),
+                    });
+                }
+            } else if overridden_field_is_referenced {
```

Comment:
I see that the "Handle override label" case and this `overridden_field_is_referenced` (along with the final `else` branch) are not mutually exclusive in JS. I wonder why that is if-chained in Rust?